### ISSUE TITLE
Option to disable gutter decorators

### ISSUE THREAD

**Issue by tarwich** [codescene-oss/codescene-vscode#21]
I am fixing an old codebase with lots of code smell. My gutter is one solid color. The problem is I can't see other indictors in the gutter, such as ErrorLens or even git modifications. I would like to disable the gutter decorations entirely, while other people might want to customize them.

I found an article on StackOverflow for how this can be done for other extensions. Perhaps you could add the ability to tweak the gutter for CodeScene? 
https://stackoverflow.com/a/43973209

Not very useful when it's this loud. Once I get the errors down to 2-3 per file this might be useful.
<img width="28" alt="image" src="https://github.com/codescene-oss/codescene-vscode/assets/334149/eb62a615-2b82-4bf4-886c-ef3eb71dec1b">

**Comment by tarwich** [comment]
Dude. Thanks for putting so much effort into this. I also looked and it seems you're right. That the gutter indicators come from VSCode. I later forgot about it. Let me look into this again. I'll comment today.

**Comment by tarwich** [comment]
I can see a few ways to clean up really dirty files. 

1. Have an option to disable errors but keep code lens
    1. Perhaps only if there are more than (N) errors in the file
2. Show a max of (N) errors per file
    1. Ideally the top (N) heaviest errors
3. Only show errors in the current scope (block / method / class)

What do you think about these? Is one path easier to pursue than another? Do you see any of these bringing value to users other than just me?

**Comment by sandlunds** [comment]
Good suggestions. Number 1 will be easier to implement, but I like number 2 also, especially the idea of ranking the heaviest issues. I think a ranking of issues, or indicating how heavy the penalty is very interesting, regardless of the issue with "gutter clutter". If we have that, adding an option to only add diagnostics for the heaviest errors would be pretty simple! Let me get back to you on this once we have a plan forward.

**Comment by tarwich** [comment]
I agree with everything you said. As I read your reply, you reminded me of the first extension I used: [CodeMetrics]. I'll share some of that experience here as inspiration.

[CodeMetrics]: https://marketplace.visualstudio.com/items?itemName=kisstkondoros.vscode-codemetrics

In CodeMetrics, you customize the gutter colors based on code complexity:

```json5
    ComplexityColorLow: "#4bb14f",
    ComplexityColorNormal: "#ffc208",
    ComplexityColorHigh: "#f44034",
    ComplexityColorExtreme: "#ff0000",
```

I'm not sure how they implemented that, but it seems this code is related:

https://github.com/kisstkondoros/codemetrics/blob/72a359058e8d6e2f305df7546b818846ad1be56a/src/editordecoration/EditorDecoration.ts#L183-L198

```ts
        const options: DecorationRenderOptions = {
            overviewRulerLane: OverviewRulerLane.Right,
            overviewRulerColor: color,
            before: {
                contentIconPath: this.getContentIconPath(decorationTemplate, color, size),
                margin: `${size / 2}px`,
            },
        };
        if (!decorationModeEnabled) {
            options.before = null;
        }
        if (!overviewRulerModeEnabled) {
            options.overviewRulerColor = null;
        }
        options.rangeBehavior = DecorationRangeBehavior.ClosedClosed;
        return window.createTextEditorDecorationType(options);
```
